nuxeo-runtime/nuxeo-launcher-commons/src/test/java/org/nuxeo/launcher/config/ConfigurationGeneratorTest.java

```diff
@@ -576,6 +585,37 @@ public void testReloadConfigurationWhenConfigurationFileWasEditedByAnotherGenera
         assertEquals("default,mongodb", configGenerator.getUserTemplates());
         assertEquals("default,mongodb", configGenerator.getUserConfig().getProperty(ConfigurationGenerator.PARAM_TEMPLATES_NAME));
     }
+    @Test
+    public void testCheckEncoding() throws Exception {
+        Path tempFile = Files.createTempFile("", "", PosixFilePermissions.asFileAttribute(new HashSet<>(
+                Arrays.asList(PosixFilePermission.OWNER_EXECUTE, PosixFilePermission.OWNER_READ, PosixFilePermission.OWNER_WRITE))
+        ));
+        // Test UTF8
```

Comment:
Please split these assertions into 3 separate tests.

Reply:
One test: Can the file correctly detect multiple formats .